### PR TITLE
feat(distill): extract session summary facts into distill pipeline

### LINKED ISSUE #319: Fix data race in TestUploadSessionFiles_Basic
## Bug

`internal/lfs/session_upload_test.go:80` has a data race detected by `-race` during `make test`.

## Root Cause

The test's `httptest` handler writes to a plain `map[string]string` from multiple concurrent HTTP handler goroutines without synchronization.

## Stack Trace (abbreviated)

```
WARNING: DATA RACE
Write at 0x00c000319500 by goroutine 93:
  runtime.mapassign_faststr()
  github.com/sageox/ox/internal/lfs.TestUploadSessionFiles_Basic.func1()
      internal/lfs/session_upload_test.go:80
```

## Fix

Guard the map with `sync.Mutex` or replace with `sync.Map`.

### PR DESCRIPTION
**Pull request by galexy**
## Summary

- Adds a new pre-distill fact extraction stage that reads `summary.json` from sessions in the ledger, transforms `AgentSummary` structured data into the uniform `facts.Fact` schema, and writes JSONL files to `memory/.session-facts/<YYYY-MM-DD>/` in the team context
- No LLM calls — pure data transformation from structured session data (decisions, action items, open questions, aha moments)
- Session facts merge into the daily distill alongside discussion and GitHub facts

## How it works

```mermaid
flowchart LR
    A[Ledger<br/>sessions/*/summary.json] -->|scanPendingSessions| B[sessionInput]
    B -->|sessionSummaryToFacts| C[facts.Fact]
    C -->|facts.WriteFacts| D[Team Context<br/>memory/.session-facts/date/]
    D -->|readPendingSessionFacts| E[distillDaily merge]
    E --> F[Daily Summary .md]
```

### Fact mapping

| Source | Category | Headline |
|--------|----------|----------|
| `AgentSummary.Decisions[]` | `decision` | `Decision.What` |
| `AgentSummary.ActionItems[]` | `action_item` | `ActionItem.Task` |
| `AgentSummary.OpenQuestions[]` | `open_question` | `OpenQuestion.Question` |
| `AhaMoments[]` (insight/breakthrough/synthesis) | `learning` | `AhaMoment.Highlight` |
| `AhaMoments[]` (decision) | `decision` | `AhaMoment.Highlight` |
| Session title + summary | `context` | Title |

### Dedup & state tracking

- Map-based dedup (`ProcessedSessions`) keyed by session dir name with content hash — matches the `ProcessedDiscussions` pattern
- Three-stage check: hash match + fact file → skip; not in state + fact file → skip (fresh clone recovery); otherwise → process
- Quality gate: sessions with `QualityScore > 0 && QualityScore < 0.2` are skipped

### Files

| File | Change |
|------|--------|
| `cmd/ox/distill_sessions.go` | **New** — all extraction logic (scan, transform, read-back) |
| `cmd/ox/distill_sessions_test.go` | **New** — 28 test cases |
| `cmd/ox/distill.go` | `ProcessedSessions` state field, wiring in `runDistill` + `distillDaily` |
| `cmd/ox/distill_write.go` | `.session-facts` in `ensureMemoryDirs` |
| `internal/facts/types.go` | `SourceSession = "session"` constant |

## Test plan

- [x] 28 unit tests pass covering scan dedup, quality filtering, fact mapping, date grouping, hash determinism, boundary conditions
- [x] All existing distill tests pass (no regressions)
- [x] `go build`, `go vet`, `make lint` clean (no new issues)
- [x] `make test` all green
- [x] Manual E2E: `ox distill --dry-run` shows 54 pending sessions on ox repo
- [x] Manual E2E: `ox distill` extracted 228 facts from 54 sessions, merged into daily summaries, synthesized into weekly

Closes #319

Co-Authored-By: [SageOx](https://github.com/SageOx)

## Summary by CodeRabbit

* **New Features**
  * Sessions are now automatically processed to extract facts including decisions, action items, insights, and open questions, which are integrated into daily distillation.
  * System tracks processed sessions using content hashing to prevent re-processing.
  * Session-derived facts are organized by date and merged into existing distillation workflows.

* **Tests**
  * Comprehensive test suite added covering session parsing, pending detection, and fact extraction logic.

### CODE REVIEW

File: cmd/ox/distill_sessions.go [hunk 1]
```diff
@@ -0,0 +1,386 @@
...
+		return nil
+	}
+
+	for _, s := range pending {
+		extractedFacts := sessionSummaryToFacts(s)
+		if len(extractedFacts) == 0 {
+			slog.Debug("no facts extracted from session", "session", s.DirName)
+			// still mark as processed to avoid re-scanning
+			state.ProcessedSessions[s.DirName] = s.Hash
+			continue
```
[the remark this answers is not in the document]
Comment (galexy) [reply cmd/ox/distill_sessions.go]: Fixed. Now writes an empty marker `.jsonl` (header-only) for zero-fact sessions so `scanPendingSessions` sees the fact file and skips on re-run. State is only updated after the marker write succeeds.
